java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -1201,38 +1075,19 @@ public void shouldThrowExceptionWithPartitionSplitRequestWhereParentIsMarkedAsLe
         Field field = new Field("key", new LongType());
         Schema schema = Schema.builder().rowKeyFields(field).build();
         StateStore dynamoDBStateStore = getStateStore(schema);
-        Partition parentPartition = dynamoDBStateStore.getAllPartitions().get(0);
-        Partition parentPartitionAfterSplit = parentPartition.toBuilder()
-                .childPartitionIds(Arrays.asList("child1", "child2"))
-                .leafPartition(true)
-                .build();
-        Region region1 = new Region(new RangeFactory(schema).createRange(field, Long.MIN_VALUE, null));
-        Partition childPartition1 = Partition.builder()
-                .rowKeyTypes(new LongType())
-                .leafPartition(true)
-                .id("child1")
-                .region(region1)
-                .childPartitionIds(new ArrayList<>())
-                .parentPartitionId("parent")
-                .build();
-        Region region2 = new Region(new RangeFactory(schema).createRange(field, Long.MIN_VALUE, null));
-        Partition childPartition2 = Partition.builder()
-                .rowKeyTypes(new LongType())
-                .leafPartition(true)
-                .id("child2")
-                .region(region2)
-                .childPartitionIds(new ArrayList<>())
-                .parentPartitionId("parent")
-                .build();
-
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .splitToNewChildren("root", "child1", "child2", Long.MIN_VALUE)
+                .buildTree();
+        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(tree.getPartition("root"), tree.getPartition("child1"), tree.getPartition("child2"));
         // When / Then
         // TODO This fails for the wrong reason. There's actually no validation of the leafPartition flag.
         //      It's failing because the parent partition ID doesn't match.
         //      We could implement the validation and improve the assertion, but we probably don't actually
         //      need validation here at all as long as the code that calls this method is well tested.
         assertThatThrownBy(() ->
                 dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(
-                        parentPartitionAfterSplit, childPartition1, childPartition2))
+                        tree.getPartition("root"), tree.getPartition("child1"), tree.getPartition("child2")))
                 .isInstanceOf(StateStoreException.class);
     }
```

Comment:
Some of the tests below here haven't been converted:

shouldThrowExceptionWithPartitionSplitRequestWhereChildrenWrong
shouldThrowExceptionWithPartitionSplitRequestWhereParentWrong
shouldThrowExceptionWithPartitionSplitRequestWhereNewPartitionIsNotLeaf

These could simulate the specific invalid data they care about by calling toBuilder on the partition to set the invalid data.